### PR TITLE
[DOP-19899] - add integration Clickhouse tests & CI

### PR DESCRIPTION
**Pull request by maxim-lixakov**
## Change Summary

- add integration Clickhouse tests
- add github actions CI to run tests

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/syncmaster/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: syncmaster/worker/handlers/db/base.py [hunk 1]
```diff
@@ -31,6 +32,12 @@ def write(self, df: DataFrame) -> None:
         writer = DBWriter(
             connection=self.connection,
             table=self.transfer_dto.table_name,
+            # TODO: get options from transfer_dto
+            options=(
+                Clickhouse.WriteOptions(createTableOptions="ENGINE = TinyLog")
```
Comment (dolfinus) [inline syncmaster/worker/handlers/db/base.py:37]: Let's discuss this in chat

File: syncmaster/worker/spark.py [hunk 2]
```diff
@@ -36,12 +36,14 @@ def get_worker_spark_session(
...
 def get_packages(db_type: str) -> list[str]:
-    from onetl.connection import Oracle, Postgres, SparkS3
+    from onetl.connection import Clickhouse, Oracle, Postgres, SparkS3
 
     if db_type == "postgres":
         return Postgres.get_packages()
     if db_type == "oracle":
         return Oracle.get_packages()
+    if db_type == "clickhouse":
+        return Clickhouse.get_packages()
```
[the remark this answers is not in the document]
Comment (maxim-lixakov) [reply syncmaster/worker/spark.py:47]: created task: https://jira.mts.ru/browse/DOP-21665

File: Makefile [hunk 3]
```diff
@@ -65,10 +65,10 @@ broker-start: ##Broker Start broker
 
 
 
-test: test-db test-broker ##@Test Run tests
+test: test-db test-broker ##@Test           Run tests
 	${POETRY} run pytest $(PYTEST_ARGS)
 
-test-db: test-db-start db-upgrade ##@TestDB Prepare database (in docker)
+test-db: test-db-start db-upgrade ##@TestDBPrepare database (in docker)
```
Comment (dolfinus) [inline Makefile:71]:
```suggestion
test-db: test-db-start db-upgrade ##@TestDB Prepare database (in docker)
```

File: syncmaster/worker/spark.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dolfinus) [reply syncmaster/worker/spark.py:47]: Please don't publish internal links on Github

File: syncmaster/worker/handlers/db/clickhouse.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                Clickhouse.WriteOptions(createTableOptions=f"ENGINE = MergeTree() ORDER BY {quoted_sort_column}")
+                if self.transfer_dto.type == "clickhouse"
+                else None
+            ),
+        )
+        return writer.run(df=normalized_df)
+
+    def normalize_column_names(self, df: DataFrame) -> DataFrame:
+        for column_name in df.columns:
+            df = df.withColumnRenamed(column_name, column_name.upper())
```
Comment (dolfinus) [inline syncmaster/worker/handlers/db/clickhouse.py:56]: upper or lower?

**Review by dolfinus — APPROVED** [review]

File: syncmaster/worker/handlers/db/clickhouse.py (hunk 4 above)
Comment (maxim-lixakov) [reply syncmaster/worker/handlers/db/clickhouse.py:56]: actually doesn't make a difference, but for now it is inconsistent in syncamaster, some column are lower ([hive](https://github.com/MobileTeleSystems/syncmaster/blob/develop/syncmaster/worker/handlers/db/hive.py#L36), [postgres](https://github.com/MobileTeleSystems/syncmaster/blob/develop/syncmaster/worker/handlers/db/postgres.py#L37), some are upper: [oracle](https://github.com/MobileTeleSystems/syncmaster/blob/develop/syncmaster/worker/handlers/db/oracle.py#L38)
what is the best version: to lower all columns or to upper?

Comment (maxim-lixakov) [reply syncmaster/worker/handlers/db/clickhouse.py]: lower case 👍

**Pull request merged**